samples/CGCatalog/CGCatalog/CGCBaseViewController.m

```diff
@@ -20,17 +20,49 @@ @interface CGCBaseViewController ()
 
 @end
 
-@implementation CGCBaseViewController
+@implementation CGCBaseViewController {
+    BOOL _drawComparisonImage;
+}
 
 - (instancetype)init {
     self = [super init];
     if (self) {
         self.title = [[NSStringFromClass([self class]) stringByReplacingOccurrencesOfString:@"CGC" withString:@""]
             stringByReplacingOccurrencesOfString:@"ViewController"
                                       withString:@""];
+        _drawComparisonImage = NO;
     }
     return self;
 }
+
+- (instancetype)initWithLineWidth:(CGFloat)width
+                            color:(CGColorRef)colorRef
+                      dashPattern:(CGFloat*)pattern
+                            phase:(CGFloat)phase
+                        dashCount:(size_t)count {
```

Comment:
Refactor these line and drawing options into a container otherwise this is going to get ridiculous passing around all these individual values.